policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/service/PolicyServiceAsyncProxy.java

```diff
@@ -48,22 +55,63 @@ public PolicyServiceAsyncProxy(final PolicyServiceHierarchyProxy service, final
+                        CompletableFuture.supplyAsync(() -> service.getResourceRules(request.getResource()), executor)
+                                .thenApply(rules -> AuditablePolicyResourceResponse.Builder.create()
+                                        .withPolicyRequest(request)
+                                        .withRules(rules)
+                                        .withNoErrors()
+                                        .withNoNoModifiedResponse())
+                                .exceptionally(e -> AuditablePolicyResourceResponse.Builder.create()
+                                        .withPolicyRequest(request)
+                                        .withRules(null)
+                                        .withAuditErrorMessage(AuditErrorMessage.Builder.create(request, Collections.emptyMap()).withError(e))
+                                        .withNoNoModifiedResponse()
+                                )
+                )
+                .orElse(CompletableFuture.completedFuture(AuditablePolicyResourceResponse.Builder.create()
+                        .withPolicyRequest(null)
+                        .withRules(null)
+                        .withNoErrors().withNoNoModifiedResponse()));
     }
 
     /**
+     * Given a nullable request, unwrap and store the request if it is non-null, ignore it if it is null
      * Async call to the {@link PolicyServiceHierarchyProxy} getRecordRules that takes a resource and gets the record rules applied against it
      *
-     * @param resource the resource the user wants access to
+     * @param modifiedAuditable the resource the user wants access to
      * @return the record rules for the LeafResource, if null then an exception will be thrown
      */
-    public CompletableFuture<Rules<Serializable>> getRecordRules(final LeafResource resource) {
-        return CompletableFuture.supplyAsync(() -> service.getRecordRules(resource), executor);
+    public CompletableFuture<AuditablePolicyRecordResponse> getRecordRules(final AuditablePolicyResourceResponse modifiedAuditable) {
+
+       PolicyRequest nullableRequest = modifiedAuditable.getPolicyRequest();
+        return Optional.ofNullable(nullableRequest)
+                .map((PolicyRequest request) ->
+                        CompletableFuture.supplyAsync(() -> service.getRecordRules(request.getResource()))
+                                .thenApply(rules -> AuditablePolicyRecordResponse.Builder.create()
+                                        .withPolicyResponse(PolicyResponse.Builder.create(request)
+                                                .withRules(rules))
+                                        .withNoErrors())
+                                .exceptionally(e -> AuditablePolicyRecordResponse.Builder.create()
+                                        .withPolicyResponse(PolicyResponse.Builder.create(request)
+                                                .withRules(new Rules()))
```

Comment:
If you look at the service call in this block, service.getRecordRules(Resource). with throw a NoSuchPolicyException, (rules will be a null in this case) and the follow through will be the exceptionally instead of the thenApply.  In other words, the thenApply path will always have an instance of rules.

Reply:
Okay, that makes sense.